http/forwarded_for_test.go

```diff
@@ -255,4 +257,80 @@ func TestHandler_XForwardedFor(t *testing.T) {
+		resp, err := client.RawRequest(req)
+		if err == nil {
+			t.Fatal("expected error")
+		}
+		defer resp.Body.Close()
+		buf := bytes.NewBuffer(nil)
+		buf.ReadFrom(resp.Body)
+		if !strings.Contains(buf.String(), "failed to base64 decode the client certificate: ") {
+			t.Fatalf("bad body: %v", buf.String())
+		}
+	})
+
+	// Next: test a valid (unverified) certificate being sent
+	t.Run("pass_cert", func(t *testing.T) {
+		t.Parallel()
+		testHandler := func(props *vault.HandlerProperties) http.Handler {
+			origHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(http.StatusOK)
+				w.Write([]byte(base64.StdEncoding.EncodeToString(r.TLS.PeerCertificates[0].Raw)))
+			})
+			listenerConfig := getListenerConfigForMarshalerTest(goodAddr)
+			listenerConfig.XForwardedForClientCertHeader = "X-Forwarded-Tls-Client-Cert"
+			listenerConfig.XForwardedForClientCertHeaderDecoders = "URL,BASE64"
+			return WrapForwardedForHandler(origHandler, listenerConfig)
+		}
+
+		cluster := vault.NewTestCluster(t, nil, &vault.TestClusterOptions{
+			HandlerFunc: HandlerFunc(testHandler),
+		})
+		cluster.Start()
+		defer cluster.Cleanup()
+		client := cluster.Cores[0].Client
+
+		req := client.NewRequest("GET", "/")
+		req.Headers = make(http.Header)
+		req.Headers.Set("x-forwarded-for", "5.6.7.8")
+		testcertificate := `MIIDtTCCAp2gAwIBAgIUf%2BjhKTFBnqSs34II0WS1L4QsbbAwDQYJKoZIhvcNAQEL%0ABQAwFjEUMBIGA1UEAxMLZXhhbXBsZS5jb20wHhcNMTYwMjI5MDIyNzQxWhcNMjUw%0AMTA1MTAyODExWjAbMRkwFwYDVQQDExBjZXJ0LmV4YW1wbGUuY29tMIIBIjANBgkq%0AhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAsZx0Svr82YJpFpIy4fJNW5fKA6B8mhxS%0ATRAVnygAftetT8puHflY0ss7Y6X2OXjsU0PRn%2B1PswtivhKi%2BeLtgWkUF9cFYFGn%0ASgMld6ZWRhNheZhA6ZfQmeM%2FBF2pa5HK2SDF36ljgjL9T%2BnWrru2Uv0BCoHzLAmi%0AYYMiIWplidMmMO5NTRG3k%2B3AN0TkfakB6JVzjLGhTcXdOcVEMXkeQVqJMAuGouU5%0AdonyqtnaHuIJGuUdy54YDnX86txhOQhAv6r7dHXzZxS4pmLvw8UI1rsSf%2FGLcUVG%0AB%2B5%2BAAGF5iuHC3N2DTl4xz3FcN4Cb4w9pbaQ7%2BmCzz%2BanqiJfyr2nwIDAQABo4H1%0AMIHyMB0GA1UdJQQWMBQGCCsGAQUFBwMBBggrBgEFBQcDAjAdBgNVHQ4EFgQUm%2B%2Be%0AHpyM3p708bgZJuRYEdX1o%2BUwHwYDVR0jBBgwFoAUncSzT%2F6HMexyuiU9%2F7EgHu%2Bo%0Ak5swOwYIKwYBBQUHAQEELzAtMCsGCCsGAQUFBzAChh9odHRwOi8vMTI3LjAuMC4x%0AOjgyMDAvdjEvcGtpL2NhMCEGA1UdEQQaMBiCEGNlcnQuZXhhbXBsZS5jb22HBH8A%0AAAEwMQYDVR0fBCowKDAmoCSgIoYgaHR0cDovLzEyNy4wLjAuMTo4MjAwL3YxL3Br%0AaS9jcmwwDQYJKoZIhvcNAQELBQADggEBABsuvmPSNjjKTVN6itWzdQy%2BSgMIrwfs%0AX1Yb9Lefkkwmp9ovKFNQxa4DucuCuzXcQrbKwWTfHGgR8ct4rf30xCRoA7dbQWq4%0AaYqNKFWrRaBRAaaYZ%2FO1ApRTOrXqRx9Eqr0H1BXLsoAq%2BmWassL8sf6siae%2BCpwA%0AKqBko5G0dNXq5T4i2LQbmoQSVetIrCJEeMrU%2BidkuqfV2h1BQKgSEhFDABjFdTCN%0AQDAHsEHsi2M4%2FjRW9fqEuhHSDfl2n7tkFUI8wTHUUCl7gXwweJ4qtaSXIwKXYzNj%0AxqKHA8Purc1Yfybz4iE1JCROi9fInKlzr5xABq8nb9Qc%2FJ9DIQM%2BXmk%3D`
```

Comment:
Would this work also
```
testcertificate := `-----BEGIN%20CERTIFICATE-----%0AMIIESDCCAzCgAwIBAgIUWWQIzsdICMRWpzb2yPYRJZVHaYgwDQYJKoZIhvcNAQEL%0ABQAwDzENMAsGA1UEAxMEVGVzdDAeFw0yMzA4MDIwNDE3NTZaFw0yMzA4MDIwNTE5%0AMTZaMBsxGTAXBgNVBAMTEG5vZGUxLmRvbWFpbi5jb20wggIiMA0GCSqGSIb3DQEB%0AAQUAA4ICDwAwggIKAoICAQCswKCIqCoKTvAySZnAnPPV8iGoS59xJnyYmJ4TChK0%0A1%2BhB8htli6L34Kh6aZ52YRMCXpKpXJEDmSjK3DtdivqOLYFoHPSe%2FBgs64NMT1Vp%0ApoWirh4oVCMb6ojGKHpYMczH3BzP5xAUfNPGWFqK5%2BI7YVQv7oR7gy8yR28bJz%2Bm%0A3Qpt5uyzCol5cC2HOepfbF6x44a%2BjLSVdNUbQxtL0n1RcNQgwur7o4c6F59gc1Bc%0Ao%2FTfJSPsWZ62ejaYBOVnV9SxvjNsw0TirxOKJ38tyHba1pTSISeaG57DD6LDku%2Bl%0Ailj3dPb6Fq8KudrgUVz9dfFeQHDhH%2F7q9fz0aaFlVbnYY6y7yIhW%2B1DJQLXTSacX%0AMq2OVwUYjlmx7NUS8bCVqwDiB4zUjlrAjE%2BxKtQ0s4B4lRNLsSoXKBT0F7N6sBBf%0ANP4BJSq6Zt5E3qMvPTPgWRAfT99m0tI13DfMWhPnAvu7kmiFsAMUzmD4B%2BUlb7ak%0A6qeyRglas9CXuVYfw7Y3s16zI7a8GSm%2BJfHxvhX70OsyTSh3CUP9WBCAmzC%2FY%2F8x%0A8jiCj87DFTXZkClCbPTH2tWukNeKg2eY0f8eL7zu2wKeBskKjakBb6SHBDkvGMah%0AyS%2F76TFpO1W8KABoGl%2Bu%2B%2BOOrz2eeQbQYTbRyJHPStOkOY6V%2F%2BA0TvqNlzWu12tD%0AnQIDAQABo4GPMIGMMA4GA1UdDwEB%2FwQEAwIDqDAdBgNVHSUEFjAUBggrBgEFBQcD%0AAQYIKwYBBQUHAwIwHQYDVR0OBBYEFIk038Eim%2Bvy%2BDPpobgdCUygwDcdMB8GA1Ud%0AIwQYMBaAFBved0eZwAgo2x1d8vOoVw3dLdDwMBsGA1UdEQQUMBKCEG5vZGUxLmRv%0AbWFpbi5jb20wDQYJKoZIhvcNAQELBQADggEBAJ1Ci7iU9bCApACMP73idO7zxCf7%0AGEmTsrRU5PgbOfJ3WMf0YBu6ynWPovGg2rh4N4KUs9cO0YJCmuerrZUIIQjPF%2FSa%0A42NWoAO35yogcXX6YS3ujz8eYfplh9rrQDQvDopJCpX6YeEY5oM2uEslCickiRI%2B%0AlhnedZ9tOdsJiAfheEeyEI%2BaQiZoXYYopMtmPHbcQ9jPRf54LeNlw%2FgX6o4BJV83%0AQhHcGh%2FrRYwjPE%2Farj2pUqY01b0z7zRAvn8g6Ju5qXzw6D3t507imFwEbE9WMFFe%0A24ZJVFCPjSN842GmrR56Dh3gR7TKXxstaNoSA0tmul9unu6tw8T%2Bx%2FtMzXQ%3D%0A-----END%20CERTIFICATE-----%0A`
```
with `-----BEGIN%20CERTIFICATE-----%0A` and `-----END%20CERTIFICATE-----%0A`

Reply:
It will work with NGINX. See https://github.com/hashicorp/vault/pull/17272#issuecomment-1564623194